models/app-models.js

```diff
@@ -44,4 +42,37 @@ const fetchReviewsbyId = (req) => {
     });
 };
 
-module.exports = { fetchCategories, fetchReviews, fetchReviewsbyId };
+const fetchReviewIdComments = (req) => {
+  const { params } = req;
+  return db
+    .query(
+      `
+      SELECT comments.*, 
+      CAST(COUNT(comment_id) AS INTEGER) AS comment_count 
+      FROM reviews 
+      LEFT JOIN comments ON comments.review_id = reviews.review_id 
+      WHERE reviews.review_id = $1
+      GROUP BY comments.comment_id 
+      ORDER BY comments.created_at DESC
+      `,
+      [params.review_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          message: `Try again - ID ${params.review_id} does not exist yet!!!`,
+        });
+      } else if (rows[0].comment_id === null) {
```

Comment:
oh I see why you've joined the tables now its to handle this 200 vs 404 case, we did this differently in the lectures but if its working its fine :)